src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java

```diff
@@ -223,6 +227,57 @@ public boolean postString(final Locale locale, final String component,
 		return cs.getComponentTranslation();
 	}
 
+	//	/**
+	//	 * get multiple component's translations from VIP server
+	//	 *
+	//	 * @param locale     a language tag to get the translations of it
+	//	 * @param components names of the components
+	//	 * @return a map contains all translations of the components
+	//	 */
+	//	public Map<String, Map<String, String>> getStrings(final Locale locale,
+	//			final List<String> components) {
+	//		logger.info("Start to execute TranslationMessage.getStrings of multiple components of single locale.");
+	//
+	//		if (null == locale || null == components || components.isEmpty()) {
+	//			logger.error(ConstantsMsg.WRONG_PARAMETER + "locales: {}, components: {}.", locale, components);
+	//			return new HashMap<>();
+	//		}
+	//
+	//		try {
+	//			ComponentsService cs = new ComponentsService(components, Arrays.asList(locale.toLanguageTag()));
+	//			return cs.getTranslation().values().iterator().next();
+	//		} catch (Exception e) {
+	//			logger.error(ConstantsMsg.EXCEPTION_OCCUR, e);
+	//			return new HashMap<>();
+	//		}
+	//	}
+
+	/**
+	 * get multiple component's translations from VIP server
+	 *
+	 * @param locales
+	 *            language tags to get the translations of them
+	 * @param components
+	 *            names of the components
+	 * @return a map contains all translations of the components of specified locales
+	 */
+	public Map<String, Map<String, Map<String, String>>> getStrings(final Set<String> locales,
+			final Set<String> components) {
+		logger.info("Start to execute TranslationMessage.getStrings of multiple components of multiple locales.");
+		if (null == locales || locales.isEmpty() || null == components || components.isEmpty()) {
+			logger.error(ConstantsMsg.WRONG_PARAMETER + "locales: {}, components: {}.", locales, components);
+			return new HashMap();
+		}
+
+		try {
+			ComponentsService cs = new ComponentsService(components, locales);
+			return cs.getTranslation();
```

Comment:
put the to the initial map and return the map at the end of this method.

Reply:
fixed